tests/Microsoft.Bot.Builder.TestBot/Dialogs/MainDialog.cs

```diff
@@ -14,8 +14,8 @@ namespace Microsoft.BotBuilderSamples
 {
     public class MainDialog : ComponentDialog
     {
-        protected readonly IConfiguration _configuration;
-        protected readonly ILogger _logger;
+        private readonly IConfiguration _configuration;
```

Comment:
I'm not quite sure about changing the `protected` modifier with `private` modifier, as the first one allows that any derived class can access these properties, but the second doesn't ([more info](https://stackoverflow.com/questions/614818/in-c-what-is-the-difference-between-public-private-protected-and-having-no)). Which StyleCop rule are we using here?

Reply:
This is the kind of changes the ApiCompat tool should be useful for. Does it detects this as a broken API?

Reply:
BTW, I also agree we should not modify property accessors.

Reply:
We were modifying the build when this pr was created. I just executed a manual build.

Reply:
Thanks @ParadoxARG 
Since there are no API compat issues, but changing a `protected` field to `private` can break some code (in case a user has inherited from MainDialog and accessed the `_configuration` from the child class) we want to maintain the _protectedness_.
If the only possible way to achieve this is to ignore this SA issue, let's do that adding the appropriated comment.

Reply:
Done